fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateRollup.java

```diff
@@ -0,0 +1,130 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.rewrite;
+
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.NamedExpression;
+import org.apache.doris.nereids.trees.expressions.SlotReference;
+import org.apache.doris.nereids.trees.expressions.VirtualSlotReference;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalAggregate;
+import org.apache.doris.nereids.trees.plans.logical.LogicalFilter;
+import org.apache.doris.nereids.trees.plans.logical.LogicalRepeat;
+import org.apache.doris.nereids.util.ExpressionUtils;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Set;
+
+/** EliminateRollup */
+public class EliminateRollup implements RewriteRuleFactory {
+    @Override
+    public List<Rule> buildRules() {
+        return ImmutableList.of(
+        logicalFilter(logicalAggregate(logicalProject(logicalRepeat()))).thenApply(ctx -> {
+            LogicalFilter filter = ctx.root;
+            LogicalAggregate agg = (LogicalAggregate) filter.child();
+            LogicalRepeat<Plan> repeat = (LogicalRepeat<Plan>) agg.child().child(0);
+            // check rollup only have one parameter, which grouping sets would be 2
+            if (repeat.getGroupingSets().size() != 2 || !repeat.getGroupingSets().get(1).isEmpty()) {
+                return null;
+            }
+            Set<Expression> groupNotNull = ExpressionUtils.inferNotNull(
+                    filter.getConjuncts(), agg.getOutputSet(), ctx.cascadesContext);
+            Expression groupingExpr = repeat.getGroupingSets().get(0).get(0);
+            // slots in rollup should be null reject
+            if (!ExpressionUtils.getInputSlotSet(groupNotNull).contains(groupingExpr)) {
```

Comment:
should check project only has Slot or Alias(Slot). if project has Alias(Slot) u should do replace for expressions in aggregate